frontends/web/src/routes/exchange/btcdirect.tsx

```diff
@@ -97,8 +104,86 @@ export const BTCDirect = ({ accounts, code }: TProps) => {
     }, 200);
   };
 
+  const handlePaymentRequest = useCallback(async (event: MessageEvent) => {
+    const { amount, currency } = event.data;
+
+    if (
+      typeof currency !== 'string'
+      || !['BTC', 'ETH', 'LTC', 'USDC', 'LINK'].includes(currency)
+    ) {
+      console.error('unexpected currency ' + currency);
+      alertUser('unexpected currency ' + currency);
+      return;
+    }
+
+    const address = { // TODO: should come from the backend
+      BTC: 'bc1qnkku53ue3fud8pquec7jrp4jmtn3tmqmwtelr6',
+      LTC: 'ltc1qldx5037gn4u4nu6nfrwskhtdngzxzgjefygk8r',
+      ETH: '0xE5b1F760BA4334bc311695e125861Eb5870018aD',
+      USDC: '0xE5b1F760BA4334bc311695e125861Eb5870018aD',
+      LINK: '0xE5b1F760BA4334bc311695e125861Eb5870018aD',
+    }[currency];
+
+    let txAmount: string;
+    if (currency !== 'BTC') {
+      txAmount = String(amount);
+    } else {
+      // this allows to correctly handle sats mode
+      const parsedAmount = await parseExternalBtcAmount(String(amount));
+      if (!parsedAmount.success) {
+        alertUser(t('unknownError', { errorMessage: 'Invalid amount' }));
+        return;
+      }
+      txAmount = parsedAmount.amount;
+    }
+
+    const txInput: TTxInput = {
+      address: address as string, // TODO: remove type casting 'as string' once the address is received from backend
+      amount: txAmount,
+      paymentRequest: null,
+      sendAll: 'no',
+      selectedUTXOs: [],
+      useHighestFee: true,
+    };
+
+    const txPropsal = await proposeTx(code, txInput);
+    if (txPropsal.success) {
+      const txNote = t('buy.pocket.paymentRequestNote') + ' BTC Direct'; // TODO: change to generic sell message with name placeholder 'Payment request from {name}'
+      const sendResult = await sendTx(code, txNote);
+      if (!sendResult.success && !('aborted' in sendResult)) {
+        alertUser(t('unknownError', { errorMessage: sendResult.errorMessage }));
+      }
+    } else {
+      if (txPropsal.errorCode === 'insufficientFunds') {
+        alertUser(t('buy.pocket.error.' + txPropsal.errorCode)); // TODO: using pocket error here, 'Insufficient funds on the selected account.'
+      } else if (txPropsal.errorCode) {
+        alertUser(t('send.error.' + txPropsal.errorCode));
+      } else {
+        alertUser(t('genericError'));
+      }
+    }
+  }, [code, t]);
```

Comment:
This is all a bit messy with various TODO's, but it should be pretty once we've merged the corresponding backend PR